src/main/java/org/spongepowered/api/entity/living/player/CooldownTracker.java

```diff
@@ -0,0 +1,89 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.living.player;
+
+import org.spongepowered.api.item.ItemType;
+
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+
+/**
+ * Provides access to the item cooldowns of a {@link Player}.
+ */
+public interface CooldownTracker {
+
+    /**
+     * Checks if the specified {@link ItemType} is currently on cooldown
+     * for the player.
+     *
+     * @param type The item type to check is on cooldown
+     * @return Whether or not the specified item type is cooldown
+     */
+    default boolean hasCooldown(final ItemType type) {
+        return !getCooldown(type).isPresent();
+    }
+
+    /**
+     * Gets the cooldown of the specified {@link ItemType} in ticks for the
+     * player, or empty if the the item type is currently not on cooldown.
+     *
+     * @param type The item type to get the cooldown for
+     * @return The cooldown remaining for this item type in ticks, if not
+     *     on cooldown
+     */
+    OptionalInt getCooldown(final ItemType type);
+
+    /**
+     * Resets the cooldown of the specified {@link ItemType} for the
+     * player.
+     *
+     * @param type The item type to reset the cooldown for
+     */
+    default void resetCooldown(final ItemType type) {
+        setCooldown(type, 0);
+    }
+
+    /**
+     * Sets the cooldown for the specified {@link ItemType} for the
+     * specified amount of ticks.
+     *
+     * @param type The item type to set the cooldown for
+     * @param ticks The amount of ticks to set the item type on cooldown for
+     */
+    void setCooldown(final ItemType type, final int ticks);
+
+    /**
+     * Gets the percentage of the specified {@link ItemType}'s cooldown that
+     * is remaining for the player, or empty if the the item type
+     * is currently not on cooldown.
+     *
+     * <p>If present, this value will be between 0.0 and 1.0.</p>
+     *
+     * @param type The item type to get the cooldown percentage remaining
+     * @return The percentage of cooldown remaining for the specified item type
+     */
+    OptionalDouble getPercentageRemaining(final ItemType type);
```

Comment:
This isn’t returning a percentage - you’ve specified that it’s between 0 and 1, not 0 and 100. I suppose you could use “getFractionRemaining”.

Reply:
Good point, I'll make this change and add a little to the Javadocs as well.